tools/travis/travis_linux_before_install.sh

```diff
@@ -1,6 +1,35 @@
 #!/bin/bash
 set -e
 
+if [ "$ARCH" = "freertoslwip" ]; then
+    echo -e "\r\n==Compile multithreaded version==" && echo -en 'travis_fold:start:before_install.build.multithread\\r'
+    mkdir -p build && cd build
+    cmake \
+    -DUA_ARCHITECTURE:STRING=freertosLWIP \
+    -DUA_ENABLE_PUBSUB:BOOL=OFF \
+    -DUA_ENABLE_METHODCALLS:BOOL=ON \
+    -DUA_ENABLE_NODEMANAGEMENT:BOOL=ON \
+    -DUA_ENABLE_AMALGAMATION:STRING=ON \
+    -DUA_ENABLE_PUBSUB:BOOL=ON \
+    -DUA_LOGLEVEL:STRING=600 ..
+	# Compile error ignored and does not cause any problems. Related Issues: #2887 and #2893
+    make -j || true
```

Comment:
hmm, this does not look so good. If make fails, then it's bad. Shouldn't be ignored.

First installing esp-idf and then building the stack should solve it?

Reply:
I couldn't find a way to compile open62541 library without this error and only needed files are amalgamated open62541.c/h files to build a flashable binary with xtensa toolchain, so;

```
/home/travis/build/cmbahadir/open62541/build/open62541.h:239:24: fatal error: lwip/tcpip.h: No such file or directory
 #include <lwip/tcpip.h>
```
error is ignored. 

There are discussions on #2887 and #2893.

Reply:
The problem is that he needs only the amalgamated files to then compile with the right tools. 
I see two possible solutions: 
1- The cleanest is to avoid this error by compiling open62541 properly as a library and then use the library in the project compilation (this instead of the using the amalgamated files in the project). This requires that open62541 is compiled with the xtensa tools, giving the right flags. Unfortunately I couldn't understand the "make" for the project later on. 
2- Create a CMake target in open62541 which only generates the amalgamated files and doesn't compile.

I see point 2 very specific. Point 1 is the best option, but also requires a good understanding of the building process of each part of the system.

Reply:
I would suggest to go with option 2 since build systems for each target maybe different. So rather than covering every toolchains configurations with different flags, it would be  enough to provide the the get amalgamated .c./.h files. 

And it will make the current "tcpip.h" error invisible to the user., so user will be more confident about what he is doing.

Just an opinion. :)